modules/hostap/src/supp_api.c

```diff
@@ -398,14 +380,47 @@ int z_wpa_supplicant_connect(const struct device *dev,
 	_wpa_cli_cmd_v("enable_network %d", resp.network_id);
 	_wpa_cli_cmd_v("select_network %d", resp.network_id);
 
+	return 0;
+rem_net:
+	_wpa_cli_cmd_v("remove_network %d", resp.network_id);
+out:
+	return ret;
+}
+
+/* Public API */
+int z_wpa_supplicant_connect(const struct device *dev,
+						struct wifi_connect_req_params *params)
+{
+	struct wpa_supplicant *wpa_s;
+	int ret = 0;
+
+	if (!net_if_is_admin_up(net_if_lookup_by_dev(dev))) {
+		wpa_printf(MSG_ERROR,
+			   "Interface %s is down, dropping connect",
+			   dev->name);
+		return -1;
+	}
+
+	k_mutex_lock(&wpa_supplicant_mutex, K_FOREVER);
+
+	wpa_s = get_wpa_s_handle(dev);
+	if (!wpa_s) {
+		ret = -1;
```

Comment:
Can we set `ret` to `-ENODEV` instead of `-1`?

Reply:
Sure, I guess we need to fix this at all places, is it okay if I do the cleanup together in another PR?

Reply:
Sure.